apps/site/lib/site_web/views/event_view.ex

```diff
@@ -56,45 +58,77 @@ defmodule SiteWeb.EventView do
-        ) ::
+          NaiveDateTime.t() | DateTime.t() | nil,
           String.t()
-  def render_event_duration(start_time, end_time)
+        ) :: %{date: String.t(), time: String.t()}
+  def get_formatted_date_time_map(start_time, end_time, formatter)
 
-  def render_event_duration(start_time, nil) do
+  def get_formatted_date_time_map(start_time, nil, formatter) do
     start_time
     |> maybe_shift_timezone
-    |> do_render_event_duration(nil)
+    |> do_date_time_formatting(nil, formatter)
   end
 
-  def render_event_duration(start_time, end_time) do
+  def get_formatted_date_time_map(start_time, end_time, formatter) do
     start_time
     |> maybe_shift_timezone
-    |> do_render_event_duration(maybe_shift_timezone(end_time))
+    |> do_date_time_formatting(maybe_shift_timezone(end_time), formatter)
   end
 
-  defp do_render_event_duration(start_time, nil) do
-    "#{pretty_date(start_time, "{WDshort}, {Mshort} {D}, {YYYY}")} \u2022 #{
-      format_time(start_time)
-    }"
-  end
+  @spec do_date_time_formatting(
+          NaiveDateTime.t() | DateTime.t(),
+          NaiveDateTime.t() | DateTime.t() | nil,
+          String.t()
+        ) :: %{date: String.t(), time: String.t()}
 
-  defp do_render_event_duration(
+  # If the end_time is the same date as the start_time, represent the time as a range
+  defp do_date_time_formatting(
          %{year: year, month: month, day: day} = start_time,
-         %{year: year, month: month, day: day} = end_time
+         %{year: year, month: month, day: day} = end_time,
+         formatter
        ) do
-    "#{pretty_date(start_time, "{WDshort}, {Mshort} {D}, {YYYY}")} \u2022 #{
-      format_time(start_time)
-    } - #{format_time(end_time)}"
+    %{
+      date: "#{pretty_date(start_time, formatter)}",
+      time: "#{format_time(start_time)} - #{format_time(end_time)}"
+    }
+  end
+
+  # If the end_time is nil OR a different day than the start time, then only use the start day / time
+  defp do_date_time_formatting(start_time, _end_time, formatter) do
+    # How to represent an event across multiple days in the calendar?
+    %{
+      date: "#{pretty_date(start_time, formatter)}",
+      time: "#{format_time(start_time)}"
+    }
+  end
+
+  @doc "Nicely render an event duration for the Event List, given two DateTimes."
+  @spec render_event_duration_list_view(
+          NaiveDateTime.t() | DateTime.t(),
+          NaiveDateTime.t() | DateTime.t() | nil
+        ) :: String.t()
+  def render_event_duration_list_view(start_time, nil) do
+    calendar = get_formatted_date_time_map(start_time, nil, "{WDshort}, {Mshort} {D}, {YYYY}")
+    "#{calendar.date} \u2022 #{calendar.time}"
+  end
+
+  def render_event_duration_list_view(
+        %{year: year, month: month, day: day} = start_time,
+        %{year: year, month: month, day: day} = end_time
+      ) do
+    calendar =
+      get_formatted_date_time_map(start_time, end_time, "{WDshort}, {Mshort} {D}, {YYYY}")
```

Comment:
Is `formatter` ever a different value?

Reply:
Not for the list view 👍

Reply:
Since `formatter` doesn't change that much, you could have a default formatter such as:
```elixir
@default_format "{WDshort}, {Mshort} {D}, {YYYY}"
```
Then rewrite the definition as:
```elixir
def get_formatted_date_time_map(start_time, end_time, formatter \\ @default_format)
```
That way you wouldn't have to specify the formatter value every time, only in the call that's in `_popup.html.eex`.